src/diffpy/utils/diffraction_objects.py

```diff
@@ -181,6 +181,17 @@ def __rtruediv__(self, other):
             divided.on_q[1] = other.on_q[1] / self.on_q[1]
         return divided
 
+    @property
+    def all_arrays(self):
+        return self._all_arrays
+
+    @all_arrays.setter
```

Comment:
when the user attempts to modify `all_array`, throw an exception. I've thought it might be better to use `exception` rather than `warning` since our `_set_xarrays` is doing some heavy lifting work and we want our users to use the way we want them to do?